### PR TITLE
Update stats on sync when resource exists or is asymmetric.

### PR DESCRIPTION
**Pull request by Osmose**
@mathjazz Does this look any better?

### CODE REVIEW

File: pontoon/base/models.py [hunk 1]
```diff
@@ -263,6 +263,11 @@ class Resource(models.Model):
 
     ALLOWED_EXTENSIONS = [f[0] for f in FORMAT_CHOICES] + ['pot']
 
+    @property
+    def is_asymmetric(self):
+        """Return True if this resource is in an asymmetric format."""
+        return self.format in ('dtd', 'properties', 'ini', 'inc')
```
Comment (mathjazz) [inline pontoon/base/models.py]: Bah, I forgot to mention l20n - please add it to the list.

**Comment by mathjazz** [comment]
r+wc: Please update the asymmetric. Nice work!

On a related note (and feel free to file a separate bug), we now call update_stats for each project and than Translation.delete() at the end of sync_projects, which also calls update_stats for each deletion:
https://github.com/Osmose/pontoon/blob/sync-stats-redux/pontoon/administration/management/commands/sync_projects.py#L70

Actually, we might just add a comment to https://bugzilla.mozilla.org/show_bug.cgi?id=1179954 and use it to refactor stats in general.

File: pontoon/administration/management/commands/sync_projects.py [hunk 2]
```diff
@@ -383,6 +401,7 @@ def execute_update_db(self):
 
                     if db_translation.is_dirty():
                         self.translations_to_update.append(db_translation)
+                        self.updated_locales.add(self.locales[locale_code])
```
Comment (mathjazz) [inline pontoon/administration/management/commands/sync_projects.py]: Why only here? Why not also in e.g. execute_create_db() and in the "# Any existing translations that were not approved get unapproved block"?

**Comment by mathjazz** [comment]
Another minor thing: feel free to replace "asymmetric" with a more appropriate word if you can find it. We could also say "monolingual" (or "not bilingual").

**Comment by Osmose** [comment]
> On a related note (and feel free to file a separate bug), we now call update_stats for each project and than Translation.delete() at the end of sync_projects, which also calls update_stats for each deletion:
> https://github.com/Osmose/pontoon/blob/sync-stats-redux/pontoon/administration/management/commands/sync_projects.py#L70

Queryset.delete() does a bulk delete and does not call delete on each instance, so update_stats is not called in that case.

**Comment by Osmose** [comment]
> Actually, we might just add a comment to https://bugzilla.mozilla.org/show_bug.cgi?id=1179954 and use it to refactor stats in general.

Yeah, the stats model is weird because not only is it data that we could just generate and cache instead of calculating and saving, but the model is used for things I think it shouldn't be, like controlling what resources show up in the menu.

**Pull request merged**